src/test/java/IvanFFavoriteTests.java

```diff
@@ -0,0 +1,109 @@
+import java.time.Duration;
+
+
+public class IvanFFavoriteTests extends BaseTestIvan{
+    @Test
+    public void checkFavorite() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        System.out.println("Number of favorites         "+(homePageIvan.getNumberfavourite()));
+
+    }
+
+
+    @Test
+    public void checkQueue() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        homePageIvan.playSongFromfav();
+        homePageIvan.currentQueue();
+    }
+    @Test
+    public void deleteSongFromFav() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        homePageIvan.deleteSongFromfav();
+
+    }
+    @Test
+    public void addSongToFavHeart() throws InterruptedException{
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        int initialNumberOfFav = homePageIvan.getNumberfavourite();
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        int updatedNumberOfFav = homePageIvan.getNumberfavourite();
+        System.out.println("Number of favorites before: "+initialNumberOfFav+" New Number Of Favorites:  "+updatedNumberOfFav);
+
+    }
+    @Test
+    public void downloadAllSongFromFav() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        homePageIvan.favorite();
+        homePageIvan.downloadAllSongFromFav();
+
+    }
+    @Test
+    public void downloadOneSongFromFav() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        AllSongsIvan allSongsIvan = new AllSongsIvan(getDriver());
+        homePageIvan.clickOnAllSongs();
+        allSongsIvan.addToFavoriteHeart();
+        homePageIvan.favorite();
+        Thread.sleep(2000);
+        homePageIvan.contextDownload();
+        Thread.sleep(2000);
+        Assert.assertTrue(homePageIvan.isSongDownloaded("HoliznaCC0 - Waiting On A Train.mp3"));
+    }
+    @Test
+    public void checkIfFavEmpty() throws InterruptedException {
+        LoginPageIvan loginPage = new LoginPageIvan(getDriver());
+        HomePageIvan homePageIvan = loginPage.login();
+        homePageIvan.favorite();
+        Thread.sleep(2000);
```

Comment:
**Thread. sleep is bad! It blocks the current thread and renders it unusable for further work!**
**Using Thread. sleep() frequently in an automation framework is not a good practice!**

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.


![image](https://user-images.githubusercontent.com/104761482/206839477-ac685ccd-9395-4a16-824e-02294926426f.png)